### PR TITLE
Runtime fix in hydro_tray_process

### PR DESCRIPTION
**Pull request by adacovsk**
[hotfix][bugfix]

## What this does
- Should fix runtime
- Might be scope creep, but fixed another bug where hematophage didn't initially remove nutrient
- Cleans up some code

### CODE REVIEW

File: code/modules/hydroponics/hydro_tray_process.dm [hunk 1]
```diff
@@ -183,8 +183,6 @@
 				overlays += image(icon = icon, icon_state = "over_lowwater3")
 		else if(get_toxinlevel() < TOXINLEVEL_MAX/5)
 			overlays += image(icon = icon, icon_state = "over_lowwater3")
-		else if(!seed && get_waterlevel() <= WATERLEVEL_MAX/5)
```
Comment (Kurfursten) [inline code/modules/hydroponics/hydro_tray_process.dm:186]: no more icon for low water if no plant?

Comment (adacovsk) [reply code/modules/hydroponics/hydro_tray_process.dm:186]: it's just a redundant else if, it never actually gets triggered ever

Comment (adacovsk) [reply code/modules/hydroponics/hydro_tray_process.dm:186]: can't have `if seed && !seed`

Comment (Kurfursten) [reply code/modules/hydroponics/hydro_tray_process.dm:186]: Right but you CAN move this functionality below the seed check; don't remove the functionality of knowing a tray is low on water without a plant in, that's actually a really impactful hydroponics change. I know, I play a lot of botanist.

Comment (adacovsk) [reply code/modules/hydroponics/hydro_tray_process.dm:186]: you're right, hopefully this fixes it

File: code/modules/hydroponics/hydro_tray_process.dm [hunk 2]
```diff
@@ -153,7 +153,7 @@
 	update_name() //fuck it i'll make it not happen constantly later
 
 	// Updates the plant overlay.
-	if(get_toxinlevel() > TOXINLEVEL_MAX/2)
+	if(get_toxinlevel() > get_waterlevel())
```
Comment (Kurfursten) [inline code/modules/hydroponics/hydro_tray_process.dm]: I may just misunderstand this, but isn't this completely different?

Comment (adacovsk) [reply code/modules/hydroponics/hydro_tray_process.dm]: I missed something previously regarding the behavior, and this should function better for the overlay

Comment (adacovsk) [reply code/modules/hydroponics/hydro_tray_process.dm]: I don't think it should affect it too much

Comment (adacovsk) [reply code/modules/hydroponics/hydro_tray_process.dm]: anyway, I put the part back and it should work

**Pull request merged**